xds/src/main/java/io/grpc/xds/internal/certprovider/CertificateProviderStore.java

```diff
@@ -119,13 +119,13 @@ public String toString() {
   }
 
   private final class CertProviderFactory
-      implements ReferenceCountingMap.ValueFactory<CertProviderKey, CertificateProvider> {
+      implements ReferenceCountingMap.ValueFactory<CertProviderKey, CertificateProvider, Object> {
 
     private CertProviderFactory() {
     }
 
     @Override
-    public CertificateProvider create(CertProviderKey key) {
+    public CertificateProvider create(CertProviderKey key, Object creationContext) {
```

Comment:
So you just create an _Object_ that is nothing in order to work around the API limitation?

Ahhh, I should have claimed everything I said should just be _suggestive_ as I do not have the full vision of your code. It was just what I felt like things that _could be_. You would really need to think by yourself if that would _really_ work, instead of just randomly throwing things in to overcome unforeseen issues and avoiding doing refactor for inappropriate code.

Anyway, I would say, this is very bad...